### PR TITLE
Indicate that max_token_ttl is required for OIDC auth methods

### PR DESCRIPTION
**Pull request by am-ak**
- Include an example for OIDC auth method
- correct the description of max_token_ttl that it is a required argument for OIDC auth method

### CODE REVIEW

File: docs/resources/acl_auth_method.md [hunk 1]
```diff
@@ -46,6 +46,32 @@ resource "consul_acl_auth_method" "minikube" {
...
+  "ClaimMappings": {
+    "http://consul.internal/first_name": "first_name",
+    "http://consul.internal/last_name": "last_name"
+  },
+  "ListClaimMappings": {
+    "http://consul.internal/groups": "groups"
+  }
+})
+}
+```
```
Comment (im2nguyen) [inline docs/resources/acl_auth_method.md:73]:
````suggestion
Define an `OIDC` auth method:

```hcl
resource "consul_acl_auth_method" "oidc" {
  name        = "auth0"
  type        = "oidc"
  max_token_ttl = "5m"
  config_json = jsonencode({
  "OIDCDiscoveryURL": "https://<AUTH0_DOMAIN>/",
  "OIDCClientID": "<AUTH0_CLIENT_ID>",
  "OIDCClientSecret": "<AUTH0_CLIENT_SECRET>",
  "BoundAudiences": ["<AUTH0_CLIENT_ID>"],
  "AllowedRedirectURIs": [
    "http://localhost:8550/oidc/callback",
    "http://localhost:8500/ui/oidc/callback"
  ],
  "ClaimMappings": {
    "http://consul.internal/first_name": "first_name",
    "http://consul.internal/last_name": "last_name"
  },
  "ListClaimMappings": {
    "http://consul.internal/groups": "groups"
  }
})
}
```

````

File: docs/resources/acl_auth_method.md [hunk 2]
```diff
@@ -54,8 +80,8 @@ The following arguments are supported:
 * `type` - (Required) The type of the ACL auth method.
 * `display_name` - (Optional) An optional name to use instead of the name
   attribute when displaying information about this auth method.
-* `max_token_ttl` - (Optional) The maximum life of any token created by this
-  auth method.
+* `max_token_ttl` - The maximum life of any token created by this
+  auth method. (Required) for OIDC auth method
```
Comment (im2nguyen) [inline docs/resources/acl_auth_method.md:84]:
```suggestion
* `max_token_ttl` - (Optional)The maximum life of any token created by this
  auth method. **This attribute is required for OIDC auth method.**
```

Comment (im2nguyen) [reply docs/resources/acl_auth_method.md:84]: is this true? optional for everything else, but only required for oidc?

Comment (am-ak) [reply docs/resources/acl_auth_method.md:84]: I don't know about the other auth methods, I only know for sure is required for OIDC

Comment (am-ak) [reply docs/resources/acl_auth_method.md:84]: @im2nguyen 
I just tested all other 3 Auth methods and TF didn't error on missing max_token_ttl. so It's optional for all except OIDC

Comment (am-ak) [reply docs/resources/acl_auth_method.md:84]: one small note, in this context it's an argument not an attribute

Comment (blake) [reply docs/resources/acl_auth_method.md:84]: Confirming that `max_token_ttl` is required for OIDC. <https://developer.hashicorp.com/consul/api-docs/acl/auth-methods#maxtokenttl>

> [MaxTokenTTL](https://developer.hashicorp.com/consul/api-docs/acl/auth-methods#maxtokenttl) (duration: 0s) - This specifies the maximum life of any token created by this auth method.…This value must be no smaller than 1 minute and no longer than 24 hours. Added in Consul 1.8.0.
>
> **This must be set to a nonzero value for type=oidc.**

**Comment by remilapeyre** [comment]
Hi @am-ak, thank you very much for reporting this issue. I had missed this when working on `consul_acl_auth_mode`, it is pretty rare that arguments can be optional or required based on another one.

I kept most of your implementation but migrated the `consul_acl_auth_method` documentation to the tfplugindocs framework as this is an ongoing migration that I do as we update the documentation of each resources. This is what most of my commits are about, sorry for the noise.

I also updated the example in the documentation to use [the one in the Consul documentation](https://developer.hashicorp.com/consul/docs/security/acl/auth-methods/oidc#sample-config).

Thanks for your contribution!

File: docs/resources/acl_auth_method.md (hunk 2 above)
Comment (remilapeyre) [reply docs/resources/acl_auth_method.md]: Thanks for the confirmation!

**Review by remilapeyre — APPROVED** [review]

**Pull request merged**